File: schemaorggo/orderaction.go
package schemaorggo

import "encoding/json"

// OrderAction see : https://schema.org/OrderAction
type OrderAction struct {
	TradeAction

	typeContext

	// DeliveryMethod see : https://schema.org/deliveryMethod
	// A sub property of instrument. The method of delivery.
	// types : DeliveryMethod
	DeliveryMethod []*DeliveryMethod `json:"deliveryMethod,omitempty"`
}

func (v OrderAction) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.TradeAction.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.DeliveryMethod
		if len(v.DeliveryMethod) == 1 {
			value = v.DeliveryMethod[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["deliveryMethod"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v OrderAction) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "OrderAction"

	return data, nil
}

func (v OrderAction) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/order_action.php
<?php

class OrderAction extends TradeAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'OrderAction';
	
	/**
	 * A sub property of instrument. The method of delivery.
	 * see : https://schema.org/deliveryMethod
	 * @var \DeliveryMethod|\DeliveryMethod[]
	 */
	public var $delivery_method;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'OrderAction'
		);
		
		$serialized = so_json_serialize( $this->delivery_method );
		if ( ! empty( $serialized ) ) {
			$out['deliveryMethod'] = $serialized;
		}
		
		return $out;
	}
}
